From 3a16e8b24b54a51b88de41b4624ed78e4852bc15 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Fri, 21 Oct 2022 10:13:06 +0200
Subject: [PATCH 0530/1695] plat-stm32mp1: conf: enable power management on
 stm32mp1

Add the power management support on STM32MP1 platform.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 570dc8ea6af4409bec19c778f09fef1cf961211a plat-stm32mp1: conf: enable power management on stm32mp1]
Change-Id: I033ce22da62a2377b155bbfc04b8291fbddb2062
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/358376
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 8120e4d0e..575f3d407 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -147,6 +147,7 @@ $(call force,CFG_DRIVERS_PINCTRL,y)
 $(call force,CFG_DRIVERS_REGULATOR,y)
 $(call force,CFG_GIC,y)
 $(call force,CFG_INIT_CNTVOFF,y)
+$(call force,CFG_PM_ARM32,y)
 $(call force,CFG_PSCI_ARM32,y)
 $(call force,CFG_REGULATOR_FIXED,y)
 $(call force,CFG_SECURE_TIME_SOURCE_CNTPCT,y)
@@ -165,6 +166,7 @@ $(call force,CFG_STM32_GPIO,y)
 $(call force,CFG_STM32_VREFBUF,y)
 $(call force,CFG_STM32MP_CLK_CORE,y)
 $(call force,CFG_STM32MP1_OPTEE_IN_SYSRAM,n)
+$(call force,CFG_STM32MP1_SCMI_SIP,n)
 $(call force,CFG_STM32MP1_SHARED_RESOURCES,n)
 $(call force,CFG_STM32MP1_RSTCTRL,y)
 $(call force,CFG_STM32MP13_CLK,y)
@@ -178,6 +180,7 @@ endif # CFG_STM32MP13
 ifeq ($(CFG_STM32MP15),y)
 $(call force,CFG_BOOT_SECONDARY_REQUEST,y)
 $(call force,CFG_DRIVERS_CLK_FIXED,y)
+$(call force,CFG_DDR_LOWPOWER,y)
 $(call force,CFG_HALT_CORES_ON_PANIC_SGI,15)
 $(call force,CFG_SECONDARY_INIT_CNTFRQ,y)
 $(call force,CFG_STM32MP1_SHARED_RESOURCES,y)
@@ -362,6 +365,9 @@ endif
 CFG_DRIVERS_NVMEM ?= y
 CFG_STM32_TAMP_NVRAM ?= y
 
+# Default use stm32mp1 PM mailbox context version 3
+CFG_STM32MP1_PM_CONTEXT_VERSION ?= 3
+
 ifneq ($(CFG_WITH_SOFTWARE_PRNG),y)
 CFG_HWRNG_PTA ?= y
 endif
-- 
2.47.2

